packages/dcos-integration-test/extra/test_helpers.py

```diff
@@ -1,5 +1,133 @@
+import copy
 import json
+import uuid
 
+from dcos_test_utils import marathon
 
+TEST_APP_NAME_FMT = 'integration-test-{}'
+
+# make the expanded config available at import time to allow determining
+# which tests should run before the test suite kicks off
 with open('/opt/mesosphere/etc/expanded.config.json', 'r') as f:
     expanded_config = json.load(f)
+
+
+def marathon_test_app(
```

Comment:
Turns out this is much harder in practice for a few reasons:
* container and networking sections of the definition are split across multiples branches
* container and networking sections require almost all of the main function parameters anyway
* some parts of the definition are lists of dicts which means a top-level update will overwrite the entire list, not update the index elements

So, we shall leave untangling this logic for another day.